### ISSUE TITLE
Unable to debug Blazor from Vscode

### ISSUE THREAD

**Issue by filipforsstrom** [dotnet/aspnetcore#46222]
### Is there an existing issue for this?

https://github.com/OmniSharp/omnisharp-vscode/issues/3819

### Describe the bug

On macOs and Windows debugging doesn't work on a new Blazor project in vscode. The browser will load and get stuck on `about:blank` for a few seconds and then close with this message: `Unable to launch browser: "Could not open wss://localhost:5001/_framework/debug/ws-proxy?browser=ws%3A%2F%2F127.0.0.1%3A54817%2Fdevtools%2Fbrowser%2Fa4ad1522-b732-435d-abfa-1b542322d11c"`

### Expected Behavior

Loading the Blazor app and start debugging

### Steps To Reproduce

- Using dotnet 6.0 or 7.0
- Create a new Blazor app with `dotnet new blazorwasm -o debugblazor`
- Open it with Vscode 
- Click on Yes for `Required assets to build and debug are missing from 'debugblazor'. Add them?`
- Hit F5
- Browser will get stuck on `about:blank` for a few seconds
- Browser will close and message appear: `Unable to launch browser: "Could not open wss://localhost:5001/_framework/debug/ws-proxy?browser=ws%3A%2F%2F127.0.0.1%3A54817%2Fdevtools%2Fbrowser%2Fa4ad1522-b732-435d-abfa-1b542322d11c"`

### Exceptions (if any)

Unable to launch browser: `"Could not open wss://localhost:5001/_framework/debug/ws-proxy?browser=ws%3A%2F%2F127.0.0.1%3A54817%2Fdevtools%2Fbrowser%2Fa4ad1522-b732-435d-abfa-1b542322d11c"`

### .NET Version

7.0.102

### Anything else?

.NET SDK:
 Version:   7.0.102
 Commit:    4bbdd14480

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  11.6
 OS Platform: Darwin
 RID:         osx.11.0-x64
 Base Path:   /usr/local/share/dotnet/sdk/7.0.102/

Host:
  Version:      7.0.2
  Architecture: x64
  Commit:       d037e070eb

.NET SDKs installed:
  3.1.415 [/usr/local/share/dotnet/sdk]
  3.1.416 [/usr/local/share/dotnet/sdk]
  3.1.419 [/usr/local/share/dotnet/sdk]
  3.1.420 [/usr/local/share/dotnet/sdk]
  3.1.421 [/usr/local/share/dotnet/sdk]
  3.1.422 [/usr/local/share/dotnet/sdk]
  3.1.424 [/usr/local/share/dotnet/sdk]
  3.1.425 [/usr/local/share/dotnet/sdk]
  3.1.426 [/usr/local/share/dotnet/sdk]
  6.0.100 [/usr/local/share/dotnet/sdk]
  6.0.101 [/usr/local/share/dotnet/sdk]
  6.0.202 [/usr/local/share/dotnet/sdk]
  6.0.300 [/usr/local/share/dotnet/sdk]
  6.0.301 [/usr/local/share/dotnet/sdk]
  6.0.400-preview.22330.6 [/usr/local/share/dotnet/sdk]
  6.0.400 [/usr/local/share/dotnet/sdk]
  6.0.402 [/usr/local/share/dotnet/sdk]
  6.0.403 [/usr/local/share/dotnet/sdk]
  6.0.404 [/usr/local/share/dotnet/sdk]
  6.0.405 [/usr/local/share/dotnet/sdk]
  7.0.100 [/usr/local/share/dotnet/sdk]
  7.0.101 [/usr/local/share/dotnet/sdk]
  7.0.102 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.21 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.22 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.25 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.26 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.27 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.28 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.30 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.31 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.4 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.6 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.8 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.12 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.13 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.21 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.22 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.25 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.26 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.27 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.28 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.30 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.31 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.8 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.12 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.13 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

**Comment by mkArtakMSFT** [comment]
@thaystg do you know what may be going on here? Can you take a look please? Thanks!

**Comment by thaystg** [comment]
I think this is related to this PR: https://github.com/dotnet/razor/pull/7929
And we don't know when this will be shipped, or if it's already shipped. Can you check the answer for it, please?

**Comment by TanayParikh** [comment]
Thanks @thaystg, that hasn't yet shipped.

**Comment by TanayParikh** [comment]
The fix has now been released in v1.25.4

**Comment by thaystg** [comment]
> The fix has now been released in v1.25.4

Tanay, if I look in this page:
https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.blazorwasm-companion

The version is not updated there yet:
![image](https://user-images.githubusercontent.com/4503299/214984144-84a66769-b05b-460f-94ae-6729d62430b2.png)


Am I looking in the wrong place?

**Comment by TanayParikh** [comment]
Hmm it looks like the C# extension was updated but the blazor wasm debugging needs to be updated separately. cc/ @Nick-Stanton

**Comment by xevoryn** [comment]
Is there an ETA for when the update of the extension will be published? It is really getting frustrating to not being able to debug wasm properly in vscode. I'm working on arch and it's the same issue.

**Comment by Nick-Stanton** [comment]
@thaystg @TanayParikh shipped!

**Comment by xevoryn** [comment]
Thanks very much but for me the exact same issue still persists after updating the extension and restarting vscode.
I am currently running arch Linux with the AUR build of vscode.

Could someone please confirm if this is only a problem while running linux?

**dotnet --info**
```
.NET SDK:
 Version:   7.0.102
 Commit:    4bbdd14480

Runtime Environment:
 OS Name:     arch
 OS Version:  
 OS Platform: Linux
 RID:         arch-x64
 Base Path:   /usr/share/dotnet/sdk/7.0.102/

Host:
  Version:      7.0.2
  Architecture: x64
  Commit:       d037e070eb

.NET SDKs installed:
  7.0.102 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/usr/share/dotnet]

global.json file:
  Not found

```

(Maybe reopen the ticket?)

**Comment by Nick-Stanton** [comment]
@xevoryn thank you for following up. Did you add the URL of your app to your `.vscode/launch.json` file?

Please see the below doc pages to see if they get you unblocked.
[Debug Blazor WebAssembly in VSCode](https://learn.microsoft.com/aspnet/core/blazor/debug?view=aspnetcore-7.0&tabs=visual-studio-code)
[Tooling for Blazor on Linux](https://learn.microsoft.com/aspnet/core/blazor/tooling?view=aspnetcore-7.0&pivots=linux)

**Comment by xevoryn** [comment]
Ok, right. I forgot about that but there is another problem. After the browser started correctly there is this message within the dev console (breakpoints are also not beeing hit):

![image](https://user-images.githubusercontent.com/76197715/216724620-5d55290d-ba95-4469-ab68-86d4eb882b75.png)

But this is in fact another issue I guess.

**Comment by Nick-Stanton** [comment]
@xevoryn are your breakpoints in `Program.cs` or an `OnInitialized{Async}` lifecycle method on the first page requested by the app? Please note that these will not hit as they are part of app startup and run before the debug proxy attaches. The error in your Chrome console is strange and one I haven't seen before. I will open a new issue and tag you in it while it gets investigated further.

**Comment by thaystg** [comment]
> Ok, right. I forgot about that but there is another problem. After the browser started correctly there is this message within the dev console (breakpoints are also not beeing hit):
> 
> ![image](https://user-images.githubusercontent.com/76197715/216724620-5d55290d-ba95-4469-ab68-86d4eb882b75.png)
> 
> But this is in fact another issue I guess or did I miss something else?

Which browser are you running? Is it chromium?

**Comment by xevoryn** [comment]
> Which browser are you running? Is it chromium?

Yes it is.

**Comment by thaystg** [comment]
> > Which browser are you running? Is it chromium?
> 
> Yes it is.

You need to update yous nuget packages to use 7.0.3 version, debugging on chromium wasn't available until this PR:
https://github.com/dotnet/aspnetcore/pull/45341

**Comment by xevoryn** [comment]
> > > Which browser are you running? Is it chromium?
> > 
> > 
> > Yes it is.
> 
> You need to update your nuget packages to use 7.0.3 version, debugging on chromium wasn't available until this PR: #45341

Am I missing something? As far as I can see the latest version of these two packages
```
Microsoft.AspNetCore.Components.WebAssembly
Microsoft.AspNetCore.Components.WebAssembly.DevServer
```
on [nuget.org](https://www.nuget.org/packages/Microsoft.AspNetCore.Components.WebAssembly#versions-body-tab) is `7.0.2`. These are the only two packages I am using. You can see that in the demo project I provided for the new issue: https://github.com/xevoryn/demo-vscode-debug-not-enabled
